desktop/AutoStartManager.ts

```diff
@@ -62,10 +68,43 @@ class AutoStartManager {
     );
   }
 
-  static isEnabled = () => AutoStartManager.service.isEnabled();
+  static isEnabled = () => StoreService.get(IS_AUTO_START_ENABLED);
+
+  static restoreNativeAutoLaunchStatus = async () => {
+    try {
+      const isEnabled = await AutoStartManager.service.isEnabled();
+      const isConfigEnabled = StoreService.get(IS_AUTO_START_ENABLED);
+
+      // migrate isEnabled() native status to app config
+      // @TODO delete in the next releases this migration
+      if (
+        isEnabled !== isConfigEnabled &&
+        !StoreService.get(MIGRATE_AUTO_LAUNCH_STATUS)
+      ) {
+        // finish migration
+        StoreService.set(MIGRATE_AUTO_LAUNCH_STATUS, true);
+        StoreService.set(IS_AUTO_START_ENABLED, isEnabled);
+        return;
+      }
+
+      // based on config file sync the config value and native auto launch status
+      if (isEnabled !== isConfigEnabled) {
```

Comment:
There should be only one scenario when this code will enable or disable and ignore migration, specifically when the user updates to the latest version, then rolls back to the previous version, and at some point updates to the new version. In this case, the configuration file will be the point of truth, and the migration will not be executed.